src/main/java/com/thealgorithms/searches/JumpSearch.java

```diff
@@ -41,12 +41,18 @@ public <T extends Comparable<T>> int find(T[] array, T key) {
 
         int limit = blockSize;
         // Jumping ahead to find the block where the key may be located
-        while (limit < length && key.compareTo(array[limit]) > 0) {
-            limit = Math.min(limit + blockSize, length - 1);
+        while (limit < length && key.compareTo(array[Math.min(limit, length - 1)]) > 0) {
```

Comment:
Could you please add your test case, with infinite loop, to `com.thealgorithms.searches.JumpSearchTest`